<!--REQUEST_ID=2949a17c-858e-4d7c-b85c-8b4d927faa10-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.47391&attempt_id=de0468eb-62fe-4f06-b58b-6110e33b5ecf)

Attempted: octave

<details><summary>Partial log (click to expand)</summary><p>

```
checking whether SDOT is called correctly from Fortran... yes
checking whether DDOT is called correctly from Fortran... yes
checking whether CDOTU is called correctly from Fortran... yes
checking whether ZDOTU is called correctly from Fortran... yes
checking BLAS library integer size... 4
checking for cheev_ in -lopenblas... yes
checking default size of Fortran INTEGER... 8
configure: error: your Fortran compiler must have an option for setting the default integer size to be the same size as your BLAS library uses (4 bytes).  See the file INSTALL for more information.
builder for '/nix/store/50ff0swi3x77fld8ggsyyha1p571i1ys-octave-4.4.1.drv' failed with exit code 1
error: build of '/nix/store/50ff0swi3x77fld8ggsyyha1p571i1ys-octave-4.4.1.drv' failed
```
</p></details>

--- a/pkgs/development/interpreters/octave/default.nix
+++ b/pkgs/development/interpreters/octave/default.nix
@@ -18,11 +18,11 @@ let
 in
 
 stdenv.mkDerivation rec {
-  version = "4.2.2";
+  version = "4.4.1";
   name = "octave-${version}";
   src = fetchurl {
     url = "mirror://gnu/octave/${name}.tar.gz";
-    sha256 = "0vkjfrpv7aikcn73bxqkph1qrhrdx7jqy193n8d8lwp7v2al7f3p";
+    sha256 = "15xfcx6dc7p204b92i7va2a7ygff637l370x7zjj3vzl2brd1yq9";
   };
 
   buildInputs = [ gfortran readline ncurses perl flex texinfo qhull
@@ -46,16 +46,14 @@ stdenv.mkDerivation rec {
     substituteInPlace libinterp/corefcn/help.cc \
       --replace 'Vmakeinfo_program = "makeinfo"' \
                 'Vmakeinfo_program = "${texinfo}/bin/makeinfo"'
-  ''
-  # REMOVE ON VERSION BUMP
-  # Needed for Octave-4.2.1 on darwin. See https://savannah.gnu.org/bugs/?50234
-  + stdenv.lib.optionalString stdenv.isDarwin ''
-    sed 's/inline file_stat::~file_stat () { }/file_stat::~file_stat () { }/' -i ./liboctave/system/file-stat.cc
   '';
 
   doCheck = !stdenv.isDarwin;
 
   enableParallelBuilding = true;
+
+  # See https://savannah.gnu.org/bugs/?50339
+  F77_INTEGER_8_FLAG = if openblas.blas64 then "-fdefault-integer-8" else "";
 
   configureFlags =
     [ "--enable-readline"
@@ -82,6 +80,7 @@ stdenv.mkDerivation rec {
     homepage = http://octave.org/;
     license = stdenv.lib.licenses.gpl3Plus;
     maintainers = with stdenv.lib.maintainers; [raskin];
+    description = "Scientific Pragramming Language";
     platforms = if overridePlatforms == null then
       (with stdenv.lib.platforms; linux ++ darwin)
     else overridePlatforms;

